### ISSUE TITLE
Could I see the panorama in VR mode?

### ISSUE THREAD

**Issue by tomtomtong** [robbykraft/Panorama#34]
Your codes are great. Could I also view it with VR mode? (the screen is split into two sides and in landscape mode)

**Comment by robbykraft** [comment]
Hi, I have explored this years ago, if my memories serve me well:
 - VR requires 2 different scene renderings with different camera position (IPD)
 - the panorama projected sphere didn't create the illusion of depth with 2 cameras in different places (panorama photo is only from one perspective)

it worked well in VR if the panorama was (infinite distance) the sky or the stars, with an OpenGL plane for ground and other 3D geometry in the scene. There is a lot of setup here, and maybe good to separate it into another project. I'd be interested to hear your thoughts.

**Comment by tomtomtong** [comment]
Thank you for your reply. If VR mode only supports still photos, it would be much easier. There are two cameras and two spheres. The photo for left eye will be mapped on left sphere and the photo for right eye will be mapped on right sphere. Different camera will render different sphere. Will this be much easier to implement?

**Comment by robbykraft** [comment]
if the same panorama is inside of a 2 spheres displaced from each other it makes a headachy feeling - i've tried this inside of an oculus.
are you targeting a thing like google cardboard?

**Comment by tomtomtong** [comment]
There is no displacement. As the photos already have parallax difference. You are right. I am looking something like Google Cardboard for iOS but not written in Unity and using objective-c instead.

**Comment by robbykraft** [comment]
do you mean the panorama images are different? there were 2 photos taken close to each other?

**Comment by tomtomtong** [comment]
Yes The camera position is slightly different when capturing these 2 images.

**Comment by robbykraft** [comment]
but do you see how when the viewer rotates their head, the axis between the eyes also rotates, and the cameras are in different places. you would need an infinite number of pictures in different locations

**Comment by tomtomtong** [comment]
We could use some devices like the link below to capture.
http://www.izugar.com/product/z6x3d.html

Tom Tong

2016-03-24 13:30 GMT+08:00 Robby Kraft <notifications@github.com>:

**Comment by SanPanG-K** [comment]
Your codes are great. Could I also view it with VR mode? (the screen is split into two sides and in landscape mode)

**Comment by danielCrespo01** [comment]
hello according to what I read in your post is easier to use vr mode spheres or cubes?

**Comment by skborhan** [comment]
@tomtomtong hi. I deploy this algorithm including axis, gyro rotation and translation in my 360 VR player for android. Using Open GL and it works fine. Just split to two different glviewport, render differently & its perfectly working with a very slight modification. This algorithm is ok for VR.

**Comment by skborhan** [comment]
@danielCrespo01 hi, best to determine which way you are stitching the image. Sphere is good to precision for rotation & translation calculation but corner of the image is slightly curved thats problem in Spherical Image frame. Cube is also good but easily to detect stitched part of the image. (I used both, then switched to sphere) :)
